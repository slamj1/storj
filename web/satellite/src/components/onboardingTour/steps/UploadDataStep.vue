// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="upload-data-area">
        <div class="upload-data-area__container">
            <h1 class="upload-data-area__container__title">Upload Data</h1>
            <p class="upload-data-area__container__sub-title">
                From here, you’ll set up Tardigrade to store data for your project using our S3 Gateway, Uplink CLI, or
                select from our growing library of connectors to build apps on Tardigrade.
            </p>
            <div class="upload-data-area__container__docs-area">
                <div class="upload-data-area__container__docs-area__option">
                    <h2 class="upload-data-area__container__docs-area__option__title">
                        Migrate Data from your Existing AWS buckets
                    </h2>
                    <img src="@/../static/images/onboardingTour/s3.png" alt="s3 gateway image">
                    <h3 class="upload-data-area__container__docs-area__option__sub-title">
                        S3 Gateway
                    </h3>
                    <p class="upload-data-area__container__docs-area__option__info">
                        Make the switch with Tardigrade’s S3 Gateway.
                    </p>
                    <a
                        class="upload-data-area__container__docs-area__option__link"
                        href="https://documentation.tardigrade.io/api-reference/s3-gateway"
                        target="_blank"
                    >
                        S3 Gateway Docs
                    </a>
                </div>
                <div class="upload-data-area__container__docs-area__option uplink-option">
                    <h2 class="upload-data-area__container__docs-area__option__title">
                        Upload Data from Your Local Environment
                    </h2>
                    <img src="@/../static/images/onboardingTour/uplinkcli.png" alt="uplink cli image">
                    <h3 class="upload-data-area__container__docs-area__option__sub-title">
                        Uplink CLI
                    </h3>
                    <p class="upload-data-area__container__docs-area__option__info">
                        Start uploading data from the command line.
                    </p>
                    <a
                        class="upload-data-area__container__docs-area__option__link"
                        href="https://documentation.tardigrade.io/api-reference/uplink-cli"
                        target="_blank"
                    >
                        Uplink CLI Docs
                    </a>
                </div>
                <div class="upload-data-area__container__docs-area__option">
                    <h2 class="upload-data-area__container__docs-area__option__title">
                        Use Tardigrade for your app’s storage layer
                    </h2>
                    <img src="@/../static/images/onboardingTour/connectors.png" alt="connectors image">
                    <h3 class="upload-data-area__container__docs-area__option__sub-title">
                        App Connectors
                    </h3>
                    <p class="upload-data-area__container__docs-area__option__info">
                        Integrate Tardigrade into your existing stack.
                    </p>
                    <a
                        class="upload-data-area__container__docs-area__option__link"
                        href="https://documentation.tardigrade.io/concepts/connectors"
                        target="_blank"
                    >
                        App Connectors
                    </a>
                </div>
            </div>
        </div>
        <VButton
            class="go-to-button"
            width="276px"
            height="40px"
            label="Go to Dashboard"
            :on-press="onButtonClick"
        />
        <span class="upload-data-area__support-info">
            Need help?
            <a class="upload-data-area__support-info__link" href="https://support.tardigrade.io/hc/en-us" target="_blank">
                Contact support
            </a>
        </span>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VButton from '@/components/common/VButton.vue';

import { RouteConfig } from '@/router';

@Component({
    components: {
        VButton,
    },
})

export default class UploadDataStep extends Vue {
    /**
     * Holds button click logic.
     * Ends onboarding tour and redirects to project dashboard.
     */
    public onButtonClick(): void {
        this.$router.push(RouteConfig.ProjectDashboard.path);
    }
}
</script>

<style scoped lang="scss">
    h1,
    h2,
    h3,
    p {
        margin: 0;
    }

    .upload-data-area {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: auto;
        font-family: 'font_regular', sans-serif;
        margin-top: 25px;

        &__container {
            padding: 60px;
            display: flex;
            flex-direction: column;
            align-items: center;
            background-color: rgba(255, 255, 255, 0.4);
            border-radius: 8px;
            margin-bottom: 50px;

            &__title {
                font-size: 32px;
                line-height: 39px;
                color: #1b2533;
                margin-bottom: 25px;
            }

            &__sub-title {
                font-size: 16px;
                line-height: 19px;
                color: #354049;
                margin-bottom: 35px;
                text-align: center;
                word-break: break-word;
                max-width: 800px;
            }

            &__docs-area {
                display: flex;
                align-items: center;
                justify-content: space-between;

                &__option {
                    padding: 30px;
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    border: 1px solid rgba(144, 155, 168, 0.4);
                    border-radius: 8px;

                    &__title {
                        font-size: 10px;
                        line-height: 15px;
                        text-align: center;
                        letter-spacing: 0.05em;
                        text-transform: uppercase;
                        color: #909ba8;
                        margin-bottom: 25px;
                        max-width: 181px;
                        word-break: break-word;
                    }

                    &__sub-title {
                        font-family: 'font_bold', sans-serif;
                        font-size: 16px;
                        line-height: 26px;
                        text-align: center;
                        color: #354049;
                        margin: 25px 0 5px 0;
                    }

                    &__info {
                        font-size: 14px;
                        line-height: 17px;
                        text-align: center;
                        color: #61666b;
                        max-width: 181px;
                        word-break: break-word;
                        margin-bottom: 15px;
                    }

                    &__link {
                        width: 181px;
                        height: 40px;
                        border: 1px solid #2683ff;
                        border-radius: 6px;
                        background-color: #fff;
                        color: #2683ff;
                        display: flex;
                        align-items: center;
                        justify-content: center;

                        &:hover {
                            background-color: #2683ff;
                            color: #fff;
                        }
                    }
                }
            }
        }

        &__support-info {
            margin-top: 50px;
            font-family: 'font_bold', sans-serif;
            font-size: 15px;
            line-height: 22px;
            text-align: center;
            color: #909ba8;

            &__link {
                text-decoration: underline;
            }
        }
    }

    .uplink-option {
        margin: 0 45px;
    }

    @media screen and (max-width: 1350px) {

        .uplink-option {
            margin: 0 15px;
        }
    }
</style>