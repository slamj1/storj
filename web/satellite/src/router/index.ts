// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router, { RouteRecord } from 'vue-router';

import AccountArea from '@/components/account/AccountArea.vue';
import AccountBilling from '@/components/account/billing/BillingArea.vue';
import BillingHistory from '@/components/account/billing/billingHistory/BillingHistory.vue';
import SettingsArea from '@/components/account/SettingsArea.vue';
import ApiKeysArea from '@/components/apiKeys/ApiKeysArea.vue';
import Page404 from '@/components/errors/Page404.vue';
import OnboardingTourArea from '@/components/onboardingTour/OnboardingTourArea.vue';
import ProjectDashboard from '@/components/project/ProjectDashboard.vue';
import ProjectMembersArea from '@/components/team/ProjectMembersArea.vue';

import store from '@/store';
import { NavigationLink } from '@/types/navigation';
const DashboardArea = () => import('@/views/DashboardArea.vue');
const ForgotPassword = () => import('@/views/forgotPassword/ForgotPassword.vue');
const LoginArea = () => import('@/views/login/LoginArea.vue');
const RegisterArea = () => import('@/views/register/RegisterArea.vue');

Vue.use(Router);

/**
 * RouteConfig contains information about all routes and subroutes
 */
export abstract class RouteConfig {
    // root paths
    public static Root = new NavigationLink('/', 'Root');
    public static Login = new NavigationLink('/login', 'Login');
    public static Register = new NavigationLink('/register', 'Register');
    public static ForgotPassword = new NavigationLink('/forgot-password', 'Forgot Password');
    public static Account = new NavigationLink('/account', 'Account');
    public static ProjectDashboard = new NavigationLink('/project-dashboard', 'Dashboard');
    public static Team = new NavigationLink('/project-members', 'Team');
    public static ApiKeys = new NavigationLink('/api-keys', 'API Keys');
    public static OnboardingTour = new NavigationLink('/onboarding-tour', 'Onboarding Tour');

    // child paths
    public static Settings = new NavigationLink('settings', 'Settings');
    public static Billing = new NavigationLink('billing', 'Billing');
    public static BillingHistory = new NavigationLink('billing-history', 'Billing History');
    // TODO: disabled until implementation
    // public static Referral = new NavigationLink('referral', 'Referral');

    // not in project yet
    // public static Referral = new NavigationLink('//ref/:ids', 'Referral');
}

export const notProjectRelatedRoutes = [
    RouteConfig.Login.name,
    RouteConfig.Register.name,
    RouteConfig.ForgotPassword.name,
    RouteConfig.Billing.name,
    RouteConfig.BillingHistory.name,
    RouteConfig.Settings.name,
    // RouteConfig.Referral.name,
];

export const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Login.path,
            name: RouteConfig.Login.name,
            component: LoginArea,
        },
        {
            path: RouteConfig.Register.path,
            name: RouteConfig.Register.name,
            component: RegisterArea,
        },
        {
            path: RouteConfig.ForgotPassword.path,
            name: RouteConfig.ForgotPassword.name,
            component: ForgotPassword,
        },
        {
            path: RouteConfig.Root.path,
            meta: {
                requiresAuth: true,
            },
            component: DashboardArea,
            children: [
                {
                    path: RouteConfig.Account.path,
                    name: RouteConfig.Account.name,
                    component: AccountArea,
                    children: [
                        {
                            path: RouteConfig.Settings.path,
                            name: RouteConfig.Settings.name,
                            component: SettingsArea,
                        },
                        {
                            path: RouteConfig.Billing.path,
                            name: RouteConfig.Billing.name,
                            component: AccountBilling,
                        },
                        {
                            path: RouteConfig.BillingHistory.path,
                            name: RouteConfig.BillingHistory.name,
                            component: BillingHistory,
                        },
                        // {
                        //     path: RouteConfig.Referral.path,
                        //     name: RouteConfig.Referral.name,
                        //     component: ReferralArea,
                        // },
                    ],
                },
                {
                    path: RouteConfig.ProjectDashboard.path,
                    name: RouteConfig.ProjectDashboard.name,
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Root.path,
                    name: 'default',
                    component: ProjectDashboard,
                },
                {
                    path: RouteConfig.Team.path,
                    name: RouteConfig.Team.name,
                    component: ProjectMembersArea,
                },
                {
                    path: RouteConfig.ApiKeys.path,
                    name: RouteConfig.ApiKeys.name,
                    component: ApiKeysArea,
                },
                {
                    path: RouteConfig.OnboardingTour.path,
                    name: RouteConfig.OnboardingTour.name,
                    component: OnboardingTourArea,
                },
            ],
        },
        {
            path: '*',
            name: '404',
            component: Page404,
        },
    ],
});

router.beforeEach((to, from, next) => {
    if (navigateToDefaultSubTab(to.matched, RouteConfig.Account)) {
        next(RouteConfig.Account.with(RouteConfig.Billing).path);

        return;
    }

    if (to.name === 'default') {
        next(RouteConfig.ProjectDashboard.path);

        return;
    }

    next();
});

router.afterEach(({name}, from) => {
    if (!name) {
        return;
    }

    if (notProjectRelatedRoutes.includes(name)) {
        document.title = `${router.currentRoute.name} | ${store.state.appStateModule.satelliteName}`;

        return;
    }

    const selectedProjectName = store.state.projectsModule.selectedProject.name ?
        `${store.state.projectsModule.selectedProject.name} | ` : '';

    document.title = `${selectedProjectName + router.currentRoute.name} | ${store.state.appStateModule.satelliteName}`;
});

/**
 * if our route is a tab and has no sub tab route - we will navigate to default subtab.
 * F.E. /account/ -> /account/billing/;
 * @param routes - array of RouteRecord from vue-router
 * @param next - callback to process next route
 * @param tabRoute - tabNavigator route
 */
function navigateToDefaultSubTab(routes: RouteRecord[], tabRoute: NavigationLink): boolean {
    return routes.length === 2 && (routes[1].name as string) === tabRoute.name;
}
