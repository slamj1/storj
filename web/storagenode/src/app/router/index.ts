// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Router from 'vue-router';

import Page404 from '@/app/components/errors/Page404.vue';

import { NavigationLink } from '@/app/types/navigation';
import DashboardArea from '@/app/views/DashboardArea.vue';
import NotificationsArea from '@/app/views/NotificationsArea.vue';
import PayoutArea from '@/app/views/PayoutArea.vue';

Vue.use(Router);

export abstract class RouteConfig {
    public static Root = new NavigationLink('', 'Root');
    public static Notifications = new NavigationLink('/notifications', 'Notifications');
    public static Payout = new NavigationLink('/payout-information', 'Payout');
}

/**
 * Router describes location mapping with components.
 */
export const router = new Router({
    mode: 'history',
    routes: [
        {
            path: RouteConfig.Root.path,
            name: RouteConfig.Root.name,
            component: DashboardArea,
        },
        {
            path: RouteConfig.Notifications.path,
            name: RouteConfig.Notifications.name,
            component: NotificationsArea,
        },
        {
            path: RouteConfig.Payout.path,
            name: RouteConfig.Payout.name,
            component: PayoutArea,
        },
        {
            path: '*',
            name: '404',
            component: Page404,
        },
    ],
});
