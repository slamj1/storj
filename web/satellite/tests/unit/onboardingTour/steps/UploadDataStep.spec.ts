// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import sinon from 'sinon';

import UploadDataStep from '@/components/onboardingTour/steps/UploadDataStep.vue';

import { router } from '@/router';
import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('UploadDataStep.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = mount(UploadDataStep, {
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('button click works correctly', async (): Promise<void> => {
        const clickSpy = sinon.spy();
        const wrapper = mount(UploadDataStep, {
            localVue,
            router,
            methods: {
                onButtonClick: clickSpy,
            },
        });

        await wrapper.find('.go-to-button').trigger('click');

        expect(clickSpy.callCount).toBe(1);
    });
});
