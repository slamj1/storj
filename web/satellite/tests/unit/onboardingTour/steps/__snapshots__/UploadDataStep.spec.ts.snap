// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDataStep.vue renders correctly 1`] = `
<div class="upload-data-area">
  <div class="upload-data-area__container">
    <h1 class="upload-data-area__container__title">Upload Data</h1>
    <p class="upload-data-area__container__sub-title">
      From here, you’ll set up Tardigrade to store data for your project using our S3 Gateway, Uplink CLI, or
      select from our growing library of connectors to build apps on Tardigrade.
    </p>
    <div class="upload-data-area__container__docs-area">
      <div class="upload-data-area__container__docs-area__option">
        <h2 class="upload-data-area__container__docs-area__option__title">
          Migrate Data from your Existing AWS buckets
        </h2> <img src="@/../static/images/onboardingTour/s3.png" alt="s3 gateway image">
        <h3 class="upload-data-area__container__docs-area__option__sub-title">
          S3 Gateway
        </h3>
        <p class="upload-data-area__container__docs-area__option__info">
          Make the switch with Tardigrade’s S3 Gateway.
        </p> <a href="https://documentation.tardigrade.io/api-reference/s3-gateway" target="_blank" class="upload-data-area__container__docs-area__option__link">
          S3 Gateway Docs
        </a>
      </div>
      <div class="upload-data-area__container__docs-area__option uplink-option">
        <h2 class="upload-data-area__container__docs-area__option__title">
          Upload Data from Your Local Environment
        </h2> <img src="@/../static/images/onboardingTour/uplinkcli.png" alt="uplink cli image">
        <h3 class="upload-data-area__container__docs-area__option__sub-title">
          Uplink CLI
        </h3>
        <p class="upload-data-area__container__docs-area__option__info">
          Start uploading data from the command line.
        </p> <a href="https://documentation.tardigrade.io/api-reference/uplink-cli" target="_blank" class="upload-data-area__container__docs-area__option__link">
          Uplink CLI Docs
        </a>
      </div>
      <div class="upload-data-area__container__docs-area__option">
        <h2 class="upload-data-area__container__docs-area__option__title">
          Use Tardigrade for your app’s storage layer
        </h2> <img src="@/../static/images/onboardingTour/connectors.png" alt="connectors image">
        <h3 class="upload-data-area__container__docs-area__option__sub-title">
          App Connectors
        </h3>
        <p class="upload-data-area__container__docs-area__option__info">
          Integrate Tardigrade into your existing stack.
        </p> <a href="https://documentation.tardigrade.io/concepts/connectors" target="_blank" class="upload-data-area__container__docs-area__option__link">
          App Connectors
        </a>
      </div>
    </div>
  </div>
  <div class="go-to-button container" style="width: 276px; height: 40px;"><span class="label">Go to Dashboard</span></div> <span class="upload-data-area__support-info">
        Need help?
        <a href="https://support.tardigrade.io/hc/en-us" target="_blank" class="upload-data-area__support-info__link">
            Contact support
        </a></span>
</div>
`;
